### PR TITLE
Data publisher integration

### PR DESCRIPTION
**Pull request by Thanu**

### CODE REVIEW

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/rule/RuleTasksDelegator.java [hunk 1]
```diff
@@ -44,347 +43,350 @@
...
+
+	public void delegateTerminateDependency(ClusterLevelPartitionContext clusterMonitorPartitionContext,
+	                                        String memberId) {
+		try {
+			//calling SM to send the instance notification event.
+			if (log.isDebugEnabled()) {
+				log.debug("delegateTerminateDependency:memberId:" + memberId);
+			}
+			//InstanceNotificationClient.getInstance().sendMemberCleanupEvent(memberId);
+			//partitionContext.moveActiveMemberToTerminationPendingMembers(memberId);
```
Comment (lahirus) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/rule/RuleTasksDelegator.java]: remove commented lines

File: components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/util/CloudControllerConstants.java [hunk 2]
```diff
@@ -24,281 +24,267 @@
...
-    // "/"+CLOUD_CONTROLLER_ELEMENT+
-    // "/"+CASSANDRA_HOST_PORT;
+	public static final String DATA_PUBLISHER_XPATH = "/" + CLOUD_CONTROLLER_ELEMENT + "/" + DATA_PUBLISHER_ELEMENT;
+	public static final String TOPOLOGY_SYNC_XPATH = "/" + CLOUD_CONTROLLER_ELEMENT + "/" + TOPOLOGY_SYNC_ELEMENT;
+	public static final String DATA_PUBLISHER_CRON_XPATH = "/" + CLOUD_CONTROLLER_ELEMENT + "/" + CRON_ELEMENT;
+	public static final String BAM_SERVER_ADMIN_USERNAME_XPATH =
+			"/" + CLOUD_CONTROLLER_ELEMENT + "/" + BAM_SERVER_ADMIN_USERNAME_ELEMENT;
+	public static final String BAM_SERVER_ADMIN_PASSWORD_XPATH =
+			"/" + CLOUD_CONTROLLER_ELEMENT + "/" + BAM_SERVER_ADMIN_PASSWORD_ELEMENT;
+	// public static final String CASSANDRA_HOST_ADDRESS_XPATH =
```
Comment (lahirus) [inline components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/util/CloudControllerConstants.java]: Remove commented lines.

**Comment by lahirus** [comment]
Is it possible to avoid jar file?( DAS-Artifacts/sparkudf-1.0-SNAPSHOT.jar)

File: DAS-Artifacts/CCEventReceiver.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (R-Rajkumar) [inline DAS-Artifacts/CCEventReceiver.xml]: don't we need to add the licence header here?

File: DAS-Artifacts/CCEventReceiver.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+<eventReceiver name="CCEventReceiver" statistics="disable"
+    trace="enable" xmlns="http://wso2.org/carbon/eventreceiver">
+    <from eventAdapterType="wso2event">
+        <property name="events.duplicated.in.cluster">false</property>
+    </from>
+    <mapping customMapping="disable" type="wso2event"/>
+    <to streamName="org.apache.stratos.cloud.controller" version="1.0.0"/>
+</eventReceiver>
```
Comment (R-Rajkumar) [inline DAS-Artifacts/CCEventReceiver.xml]: don't we need to add a licence header to this file?

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java [hunk 5]
```diff
@@ -48,194 +48,205 @@
...
+
+	private static CloudControllerServiceStub stub;
+
+	/* An instance of a CloudControllerClient is created when the class is loaded.
+	 * Since the class is loaded only once, it is guaranteed that an object of
+	 * CloudControllerClient is created only once. Hence it is singleton.
+	 */
+	private static class InstanceHolder {
+		private static final AutoscalerCloudControllerClient INSTANCE = new AutoscalerCloudControllerClient();
+	}
```
Comment (R-Rajkumar) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java]: I know this code is there from the beginning. But why do we need this inner class? IMO, we don't need an inner class to make a parent class a singleton.

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java [hunk 6]
```diff
@@ -48,194 +48,205 @@
...
+			instanceContext.setClusterInstanceId(clusterInstanceId);
+			instanceContext.setPartition(AutoscalerObjectConverter.convertPartitionToCCPartition(partition));
+			instanceContext.setInitTime(System.currentTimeMillis());
+			instanceContext.setObsoleteExpiryTime(expiryTime);
+			instanceContext.setNetworkPartitionId(networkPartitionId);
+
+			Properties memberContextProps = new Properties();
+			Property isPrimaryProp = new Property();
+			isPrimaryProp.setName("PRIMARY");
+			isPrimaryProp.setValue(String.valueOf(isPrimary));
```
Comment (R-Rajkumar) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java]: what is this PRIMARY property?

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/rule/RuleTasksDelegator.java (hunk 1 above)
Comment (Thanu) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/rule/RuleTasksDelegator.java]: It was already there that's why I didn't remove it. Now I've removed it.

File: components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/util/CloudControllerConstants.java (hunk 2 above)
Comment (Thanu) [inline components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/util/CloudControllerConstants.java]: It was already there that's why I didn't remove it. Now I've removed it.

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java (hunk 6 above)
Comment (Thanu) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java]: It was already there but it is not used. I'll remove it.

**Comment by imesh** [comment]
Thanuja: Please provide a description for your pull request explaining features that you have implemented. As I can see there is a problem in your commits, it cannot be merged:

**Comment by Thanu** [comment]
Closing this to make one commit for all the changes.

**Pull request closed without merging**